catalog/src/main/kotlin/com/adevinta/spark/catalog/examples/Navigation.kt

```diff
@@ -22,69 +22,67 @@
         ComponentsListScreen(
             components = components,
-            navController = navController,
             contentPadding = contentPadding,
+            onExampleSectionClick = {
+                navController.navigateToExampleSection(id = it)
+            },
         )
     }
 
-    composable(
-        route = "$ComponentRoute/" +
-            "{$ComponentIdArgName}",
-        arguments = listOf(
-            navArgument(ComponentIdArgName) { type = NavType.IntType },
-        ),
-    ) { navBackStackEntry ->
-        val arguments = requireNotNull(navBackStackEntry.arguments) { "No arguments" }
-        val componentId = arguments.getInt(ComponentIdArgName)
+    composable<ExampleSection> { navBackStackEntry ->
+        val exampleSection = navBackStackEntry.toRoute<ExampleSection>()
+        val componentId = exampleSection.componentId
         val component = components.first { component -> component.id == componentId }
         Component(
             component = component,
             contentPadding = contentPadding,
             onExampleClick = { example ->
                 val exampleIndex = component.examples.indexOf(example)
-                val route = "$ExampleRoute/$componentId/$exampleIndex"
-                navController.navigate(route)
+                navController.navigateToExampleShowcase(id = componentId, index = exampleIndex)
             },
         )
     }
-    composable(
-        route = "$ExampleRoute/" +
-            "{$ComponentIdArgName}/" +
-            "{$ExampleIndexArgName}",
-        arguments = listOf(
-            navArgument(ComponentIdArgName) { type = NavType.IntType },
-            navArgument(ExampleIndexArgName) { type = NavType.IntType },
-        ),
-    ) { navBackStackEntry ->
-        val arguments = requireNotNull(navBackStackEntry.arguments) { "No arguments" }
-        val componentId = arguments.getInt(ComponentIdArgName)
-        val exampleIndex = arguments.getInt(ExampleIndexArgName)
+    composable<ExampleShowcase> { navBackStackEntry ->
+        val exampleShowkase = navBackStackEntry.toRoute<ExampleShowcase>()
+        val componentId = exampleShowkase.componentId
+        val exampleIndex = exampleShowkase.exampleIndex
         val component = components.first { component -> component.id == componentId }
         val example = component.examples[exampleIndex]
-        Example(
-            example = example,
-        )
+        Example(example = example)
     }
 }
 
-internal const val ComponentRoute = "component"
-internal const val ExampleRoute = "example"
-internal const val ComponentIdArgName = "componentId"
-internal const val ExampleIndexArgName = "exampleIndex"
+private fun NavController.navigateToExampleSection(id: Int) {
+    navigate(route = ExampleSection(componentId = id))
+}
+
+private fun NavController.navigateToExampleShowcase(id: Int, index: Int) {
+    navigate(route = ExampleShowcase(componentId = id, exampleIndex = index))
+}
```

Comment:
Same here

Reply:
Fixed in https://github.com/adevinta/spark-android/pull/1401/commits/b231b1cc07ea384f7c81c9a628d7d93e900eb9a1